chains/solana/contracts/crates/common/src/lib.rs

```diff
@@ -0,0 +1,50 @@
+//! Contains common logic used across ccip programs. Common logic is versioned independently:
+//! v1 of the common logic may not necessarily be used by v1 of each particular program.
+use anchor_lang::prelude::*;
+
+pub mod seed;
+pub mod v1;
+
+#[error_code]
+pub enum CommonCcipError {
+    #[msg("The given sequence interval is invalid")]
+    // offset error code so that they don't clash with other programs
+    // (Anchor's base custom error code 6000 + offset 3000 = start at 9000)
+    InvalidSequenceInterval = 4000,
+    #[msg("Invalid pool accounts")]
+    InvalidInputsPoolAccounts,
+    #[msg("Invalid token accounts")]
+    InvalidInputsTokenAccounts,
+    #[msg("Invalid config account")]
+    InvalidInputsConfigAccounts,
+    #[msg("Invalid Token Admin Registry account")]
+    InvalidInputsTokenAdminRegistryAccounts,
+    #[msg("Invalid LookupTable account")]
+    InvalidInputsLookupTableAccounts,
+    #[msg("Invalid LookupTable account writable access")]
+    InvalidInputsLookupTableAccountWritable,
+}
+
+// Used only to enable the `account` attribute to enable account deserialization, for accounts
+// owned by the router (see TokenAdminRegistry below)
+declare_id!("Ccip842gzYHhvdDkSyi2YVCoAWPbYJoApMFzSxQroE9C");
+pub mod accounts_for_deserialization {
+    /// This mod holds structs that represent accounts owned by other programs,
+    /// but are used here just to deserialize data from them. Must be kept in sync
+    /// with their originals.
```

Comment:
Wait, if they still must be kept in sync in multiple places, does it really make sense to centralize this here?

Reply:
We're only centralizing this struct out of necessity, because it's referenced by TokenAccounts. However, since we only have this problem with a struct in `ccip_router`, we can get away with defining the "canonical" one here and losing the original. This will work fine as long as we don't have to centralize structs from other crates in here, because at that point we'd run into the need to have different top level `ID`.

Reply:
Alright, I was able to deduplicate this by moving some things around and centralizing the definition in `ccip_common`.Take a look at the new arrangement, I think it's better than before and reflects the approach we took with `base_token_pool` as Agustina suggested.